extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java

```diff
@@ -404,7 +404,7 @@ public void putRequest(final DruidPullRequest request)
         this.requestsWrite.add(request);
         if (!hasNotified) {
           hasNotified = true;
-          notify();
+          notifyAll();
```

Comment:
The class that this extends uses this pattern extensively:

```java
        synchronized (this) {
            if (!this.hasNotified) {
                this.hasNotified = true;
                this.notify();
            }
        }
```

I don't think we need to second-guess it. Maybe revert back to `notify` and add a comment about why `notifyAll` is not used?

Reply:
👍